reconcile/skupper_network/models.py

```diff
@@ -0,0 +1,291 @@
+    )
+    controller_pod_antiaffinity: str = Field(
+        Defaults.DEFAULT_CONTROLLER_POD_ANTIAFFINITY,
+        alias="controller-pod-antiaffinity",
+    )
+    controller_service_annotations: str = Field(
+        Defaults.DEFAULT_CONTROLLER_SERVICE_ANNOTATIONS,
+        alias="controller-service-annotations",
+    )
+    edge: bool = Field(Defaults.DEFAULT_EDGE, alias="edge")
+    ingress: str = Field(Defaults.DEFAULT_INGRESS, alias="ingress")
+    name: str
+    router_console: bool = Field(
+        Defaults.DEFAULT_ROUTER_CONSOLE, alias="router-console"
+    )
+    router_cpu_limit: str = Field(
+        Defaults.DEFAULT_ROUTER_CPU_LIMIT, alias="router-cpu-limit"
+    )
+    router_cpu: str = Field(Defaults.DEFAULT_ROUTER_CPU, alias="router-cpu")
+    router_memory_limit: str = Field(
+        Defaults.DEFAULT_ROUTER_MEMORY_LIMIT, alias="router-memory-limit"
+    )
+    router_memory: str = Field(Defaults.DEFAULT_ROUTER_MEMORY, alias="router-memory")
+    router_logging: str = Field(Defaults.DEFAULT_ROUTER_LOGGING, alias="router-logging")
+    router_pod_antiaffinity: str = Field(
+        Defaults.DEFAULT_ROUTER_POD_ANTIAFFINITY, alias="router-pod-antiaffinity"
+    )
+    router_service_annotations: str = Field(
+        Defaults.DEFAULT_ROUTER_SERVICE_ANNOTATIONS, alias="router-service-annotations"
+    )
+    routers: int = Field(Defaults.DEFAULT_ROUTERS, alias="routers")
+    service_controller: bool = Field(
+        Defaults.DEFAULT_SERVICE_CONTROLLER, alias="service-controller"
+    )
+    service_sync: bool = Field(Defaults.DEFAULT_SERVICE_SYNC, alias="service-sync")
+
+    class Config:
+        allow_population_by_field_name = True
+
+    @classmethod
+    def init(
+        cls,
+        name: str,
+        defaults: Optional[SkupperSiteConfigDefaultsV1] = None,
+        config: Optional[SkupperSiteConfigV1] = None,
+    ) -> SkupperConfig:
+        """Create a SkupperConfig instance by merging skupper network defaults, site configs and integration defaults."""
+        c: dict[str, Any] = {}
+
+        for field in cls.__fields__:
+            if field in ["name"]:
+                continue
+
+            c[field] = getattr(Defaults, f"DEFAULT_{field.upper()}")
+            if defaults and getattr(defaults, field, None) is not None:
+                c[field] = getattr(defaults, field)
+            if config and getattr(config, field, None) is not None:
+                c[field] = getattr(config, field)
+
+        return cls(name=name, **c)
+
+    def as_configmap_data(self) -> dict[str, str]:
+        """Return a dict with the configmap data (keys (field alias) and values as strings)."""
+        data = {}
+        for k, v in self.dict(by_alias=True).items():
+            if isinstance(v, bool):
+                # True -> "true", False -> "false"
+                v = str(v).lower()
+            data[k] = str(v)
+        return data
+
+
+class SkupperSite(BaseModel):
```

Comment:
i love this class. it holds some powerful functions to make reading the rest of the code a breeze